### DISCUSSION TITLE
r.sim.water iterations

### DISCUSSION THREAD

**Discussion by xycarto** [OSGeo/grass#4096] [Q&A] [answered]
This is not a bug per se, nor a feature request.  I am interested in determining the actual function of the `ninterations` perimeter.

In the manual, it states "_Duration of simulation is controlled by the niterations parameter. The default value is 10 minutes, reaching the steady-state may require much longer time, depending on the time step, complexity of terrain, land cover and size of the area. _" but I am a little unclear as to what the actual meaning of this is.  For example, if I wanted to simulate a 12 hour rain event at 25mm/hr, does this mean I simply set `niterations=720` and `rain_value=25`?  I know this might seem like a trivial question, but the manual is not entirely clear here.  Some other sources I have found, indicate that the 'niterations' perimeter is not the actual time of the simulation.

If it is indeed the time of simulation, perhaps the name `niterations` could be updated?

**Comment by echoix** [comment]
Why not a discussion? It's a valid question, I'm curious too ;)

**Comment by veroandreo** [comment]
Maybe @hmitaso or @baharmon can help here?

**Comment by xycarto** [comment]
Thank you for moving this over to discussions.

**Reply by hmitaso** [reply, marked as answer]
niterations is indeed duration of simulated event in minutes which is converted to number of iterations that the particles move across the study region in the code based on the average flow velocity. But the simulation stops once the steady state (peak flow) is reached which may be much sooner than after 12 hours. Number of iterations was introduced to allow users to run shorter events, to get water depths  before steady state is reached or to prevent the simulation to run for long period of time if water is just filling some bigger depression with steady state reached everywhere else.
Perhaps in next major release we can rename niterations to duration (niter was based on the original parameter where the user had to provide number of iterations instead of the more intuitive duration). I hope this explains it.

**Reply by xycarto** [reply]
Cheers @hmitaso. This was really helpful.

**Reply by neteler** [reply]
@xycarto (all): what about expanding the manual page for now? Please suggest a snippet as the discussion here get's easily "lost".
